src/probnum/filtsmooth/statespace/sde.py

```diff
@@ -0,0 +1,307 @@
+"""SDE models as transitions."""
+import numpy as np
+import scipy.linalg
+
+import probnum.random_variables as pnrv
+from probnum.filtsmooth.statespace import discrete_transition, transition
+
+
+class SDE(transition.Transition):
+    """
+    Stochastic differential equation.
+
+    .. math:: d x_t = g(t, x_t) d t + l(t, x_t) d w_t,
+
+    driven by a Wiener process with unit diffusion.
+    """
+
+    def __init__(self, driftfun, dispersionfun, jacobfun):
+        self._driftfun = driftfun
+        self._dispersionfun = dispersionfun
+        self._jacobfun = jacobfun
+
+    def transition_realization(self, real, start, stop, **kwargs):
+        raise NotImplementedError
+
+    def transition_rv(self, rv, start, stop, **kwargs):
+        raise NotImplementedError
+
+    def drift(self, time, state, **kwargs):
+        return self._driftfun(time, state, **kwargs)
+
+    def dispersion(self, time, state, **kwargs):
+        return self._dispersionfun(time, state, **kwargs)
+
+    def jacobian(self, time, state, **kwargs):
+        return self._jacobfun(time, state, **kwargs)
+
+    @property
+    def dimension(self):
+        raise NotImplementedError
+
+
+class LinearSDE(SDE):
+    """
+    Linear stochastic differential equation (SDE),
+
+    .. math:: d x_t = [G(t) x_t + v(t)] d t + L(t) x_t d w_t.
+
+    For Gaussian initial conditions, this solution is a Gaussian process.
+
+    Parameters
+    ----------
+    driftmatrixfun : callable, signature=(t, \\**kwargs)
+        This is F = F(t). The evaluations of this function are called
+        the drift(matrix) of the SDE.
+        Returns np.ndarray with shape=(n, n)
+    forcevecfun : callable, signature=(t, \\**kwargs)
+        This is u = u(t). Evaluations of this function are called
+        the force(vector) of the SDE.
+        Returns np.ndarray with shape=(n,)
+    dispmatrixfun : callable, signature=(t, \\**kwargs)
+        This is L = L(t). Evaluations of this function are called
+        the dispersion(matrix) of the SDE.
+        Returns np.ndarray with shape=(n, s)
+
+    Notes
+    -----
+    If initial conditions are Gaussian, the solution is a Gauss-Markov process.
+    We assume Gaussianity for :meth:`chapmankolmogorov`.
+    """
+
+    def __init__(self, driftmatrixfun, forcevecfun, dispmatrixfun):
+        self._driftmatrixfun = driftmatrixfun
+        self._forcevecfun = forcevecfun
+        self._dispmatrixfun = dispmatrixfun
+        super().__init__(
+            driftfun=(lambda t, x: driftmatrixfun(t) @ x + forcevecfun(t)),
+            dispersionfun=(lambda t, x: dispmatrixfun(t) @ x),
+            jacobfun=(lambda t, x: dispmatrixfun(t)),
+        )
+
+    def transition_realization(self, real, start, stop, **kwargs):
+        if "euler_step" not in kwargs.keys():
```

Comment:
not sure. I think I did it for adherence to interfaces. What do you think?

Reply:
Im again not sure about what the best practices are. It sounds reasonable to me that functions in classes and subclasses should share function arguments, so that I'd be fine with keeping it the way it is. On the other hand, if "euler_step" is actually required, then I'd also like the code to reflect that, which is most easily and obviously done by having it as an explicit argument. 

Your call :)

Reply:
I think you are right. Let's make explicit arguments in the "lowest" classes and capture those in the superclasses in the kwargs to maintain exchangability.

Reply:
screw my linter ;)